doc/contribute/guidelines.rst

```diff
@@ -554,6 +554,31 @@ exceptions:
 * Avoid using binary literals (constants starting with ``0b``).
 * Avoid using non-ASCII symbols in code, unless it significantly improves
   clarity, avoid emojis in any case.
+* Use proper capitalization of nouns in code comments (e.g. ``UART`` and not
+  ``uart`` or ``CMake`` ``cmake``).
+
+Beyond C code, the following coding style rules apply to other types of files:
```

Comment:
we need some kind of paragraph somewhere very prominently stating that this only applies to new file? or does it? If not, this doesn't suffice to solve the problem of "but I used the exact same style as file X that was the closest thing I found to what I wanted to contribute - how can this be wrong?"

Need something explaining *very* clearly how contributors are expected to deal with files that do *not* comply to these rules, cause in practice this is where a lot of personal preferences tend to pollute the review process ("It must be X because that's what we need it to be going forward!" ... "Of course it needs to be Y unlike what our guidelines say because we want to be consistent with the rest of the file/area/<whatever>", ... etc.)  

Also, many of these rules I hope we can just fix tree-wide if they are that big of an issue for the project (ex. extra blank lines in DT, etc.) otherwise there will ALWAYS be people just copy-pasting from a "broken" file and we will still need to spend cycles talking through why they shouldn't have (and no, "you didn't RTFM and all the fine print in our contribution guidelines!", won't be a good answer IMO)

Reply:
I am quite sure that we comply with these rules for most CMake, Devicetree and Kconfig files in general. What I can do, as you suggest, is to add a paragraph that states that these rules are for new files (or new code inside a file) only, so that when the user modifies an existing file they don't need to go and fix existing issues that they have not added themselves.

Reply:
> I am quite sure that we comply with these rules for most CMake, Devicetree and Kconfig files in general.

Yeah, well, not really, not even close for some of these rules, hence why I'm bringing it up :) thousands of tabs in CMakeLists.txt files, 100s of extra blank lines in DTS files, ... etc

Reply:
Regardless of the actual count, the problem does exist. I will update the text to reflect this.